docs/How_config_files_are_merged.md

```diff
@@ -0,0 +1,576 @@
+
+```
+zowe config list --locations --root
+T:\proj_config\zowe.config.user.json
+T:\proj_config\zowe.config.json
+C:\Users\YourUserName\.zowe\zowe.config.user.json
+C:\Users\YourUserName\.zowe\zowe.config.json
+```
+
+The following command shows every value obtained from every configuration file. With this command you do **NOT** see the resulting values after the configurations are merged. Instead, you see each value from each configuration file. While it requires a detailed inspection of the output, you can find an incorrect property value and know from which configuration file that value came.
+```
+zowe config list --locations
+T:\proj_config\zowe.config.user.json:
+  $schema:   ./zowe.schema.json
+  profiles:
+    YourZosmfProfileName:
+      type:       zosmf
+      properties:
+        host: YourHostname.YourCompany.com
+        port: 1234
+        user: YourUserName
+
+    ... Many more properties are displayed ...
+
+```
+<br/>
+
+## Behavior of configuration files in specific scenarios
+
+Upcoming sections describe the detailed behavior that occurs under specific deployments of configuration files. Before describing each scenario, we identify some general considerations.
+
+The behavior for merging configuration files is the same in Zowe Explorer as with Zowe CLI. However, the following behavior of Zowe Explorer can mislead you into believing that Zowe Explorer merges configuration files differently than the CLI, but the merging behavior is the same in both products.
+
+- A Zowe Explorer behavior that may appear to be a difference in merging, is actually a difference in how Zowe Explorer sets the protocol. If you set the zosmf protocol to "https" in one config file, but override the protocol with "http" in an overriding config file, the CLI will report a `socket hang up` error when the service only accepts https connections. However, Zowe Explorer will successfully perform the operation. This gives the appearance that Zowe Explorer does not override the protocol in the configuration. However, we believe that the configuration merging is fine. We believe that Zowe Explorer hard-codes the use of the "https" protocol, rather than accept the value from the profile configuration.
+
+- When a user name or password is not specified in the configuration, Zowe Explorer will prompt for the user name or password. If the configuration does not specify a default base profile, Zowe Explorer will display the errors below. The CLI runs successfully in such a configuration. To avoid the following errors in Zowe Explorer, the default "base" profile name must have a value. When `autoStore` is false, Zowe Explorer does not even try to save the values for which it prompted, so the base profile itself does not have to exist.
```

Comment:
Do we believe this is intended behavior of Zowe Explorer? Both items sound like potential bugs to me.

Reply:
I did not know what behaviors to expect, so I wrote down every result so that we have a record that we can refer to. If a customer reports such a behavior, at least we know that it is expected 'bad' behavior. The saving grace is that the 'config init' command creates a base profile and sets it as the default base profile. Thus, customers are unlikely to encounter this error. I only got the error because I had been hacking up my config file by hand, and removed the base profile as "unnecessary" for what I was doing. Apparently it was necessary.

I agree that it would be reasonable to open a GitHub issue about this behavior. I am inclined to keep this documentation until the buggy behavior is fixed.

Reply:
I created a ZE GitHub issue, and added a reference to that issue in this document.